CHANGELOG.md

```diff
@@ -4,6 +4,57 @@ This document includes a curated changelog for each release. We also publish a c
 a [GitHub release](https://github.com/nginxinc/nginx-gateway-fabric/releases), which, by contrast, is auto-generated
 and includes links to all PRs that went into the release.
 
+## Release 1.2.0
+
+*March 21, 2024*
+
+FEATURES:
+
+- NGINX Plus can now be used as the data plane. [PR-1394](https://github.com/nginxinc/nginx-gateway-fabric/pull/1394)
```

Comment:
Might be good to add a link to the N+ documentation here

Reply:
Ours or the general N+ doc?

Reply:
Ours